docs/whats-new/dotnet-aspire-9.2.md

```diff
@@ -0,0 +1,516 @@
+});
+
+var api = builder.AddProject<Projects.Api>("api")
+                .WithReference(cs);
+```
+
+### 📥 Container resources can now specify an image pull policy
+
+Container resources can now specify an `ImagePullPolicy` to control when the image is pulled. This is useful for resources that are updated frequently or that have a large image size. The following policies are supported:
+
+- `Default`: Default behavior (which is the same as missing in 9.2).
+- `Always`: Always pull the image.
+- `Missing`: Ensures the image is always pulled when the container starts.
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var cache = builder.AddRedis("cache")
+                   .WithImageTag("latest")
+                   .WithImagePullPolicy(ImagePullPolicy.Always)
+                   .WithRedisInsight();
+```
+
+The `ImagePullPolicy` is set to `Always`, which means the image will always be pulled when the resource is created. This is useful for resources that are updated frequently.
+
+### 📂 New container files API
+
+In .NET Aspire 9.2, we've added a new `WithContainerFiles` API, a way to create files and folders inside a container at runtime by defining them in code. Under the hood, it uses `docker cp` / `podman cp` to copy the files in. Supports setting contents, permissions, and ownership—no bind mounts or temp files needed.
+
+## 🤝 Integrations updates
+
+Integrations are a key part of .NET Aspire, allowing you to easily add and configure services in your app. In .NET Aspire 9.2, we've made several updates to integrations:
+
+### 🔐 Redis/Valkey/Garnet: Password support enabled by default
+
+The Redis, Valkey, and Garnet containers enable password authentication by default. This is part of our goal to be secure by default—protecting development environments with sensible defaults while still making them easy to configure. Passwords can be set explicitly or generated automatically if not provided.
+
+### 💾 Automatic database creation support
+
+There's [plenty of feedback and confusion](https://github.com/dotnet/aspire/issues/7101) around the `AddDatabase` API. The name implies that it adds a database, but it didn't actually create the database. In .NET Aspire 9.2, the `AddDatabase` API now creates a database for the following hosting integrations:
+
+| Hosting integration | API reference |
+|--|--|
+| [📦 Aspire.Hosting.SqlServer](https://www.nuget.org/packages/Aspire.Hosting.SqlServer) | <xref:Aspire.Hosting.SqlServerBuilderExtensions.AddDatabase*> |
+| [📦 Aspire.Hosting.PostgreSql](https://www.nuget.org/packages/Aspire.Hosting.PostgreSql) | <xref:Aspire.Hosting.PostgresBuilderExtensions.AddDatabase*> |
+
+The Azure SQL and Azure PostgreSQL hosting integrations both expose an `AddDatabase` API, but they don't create a database—unless you call their respective `RunAsContainer` methods. For more information, see [Understand Azure integration APIs](../azure/integrations-overview.md#understand-azure-integration-apis).
```

Comment:
@davidfowl @sebastienros - what does this line mean exactly?

Azure SQL And Azure PostgreSQL will create the database in Azure via bicep.

Reply:
Correct, the database is created. The script won't run though so if it has more than CREATE DATABASE people might get confused.

We could imagine implementing it as part of a deployment script too to seed even and Azure database when it's deployed. Migration scripts.

Reply:
I tweaked the wording here in https://github.com/dotnet/docs-aspire/pull/3066. If it needs to be adjusted, can you create a PR to change it?